neurodamus/io/synapse_reader.py

```diff
@@ -358,10 +395,14 @@ def _load_params_custom(self, _populate, _read):
                 _populate("offset", _read("morpho_offset_segment_post"))
 
     def _load_synapse_parameters(self, gid):
-        if gid not in self._data:
-            self.preload_data([gid])
+        data = self._data.get(gid, False)
+        if data is False:  # strictly compare to False (not in _data)
```

Comment:
What is the advantage of this change of strictly compare to False?

Reply:
to highlight `False` came from a missing key (line up). We must not use `if not data`, because `{}` (empty data) would match

Reply:
The benefit is that now we do only one dict lookup, not two

Reply:
Does this optimization even matter when we are doing a lot of I/O? Even when we are reading a single synapse per GID, it seems like the less confusing, cleaner lookup would not be excessively punitive.

Reply:
Once we get used to do this way it becomes kind of automatic, and the loss of readability I find minimal. How hard you want the old way?

Reply:
That's better, thanks!